### ISSUE TITLE
Empty items related columns in parent table

### ISSUE THREAD

**Issue by yolile** [open-contracting/spoonbill#67]
command:  spoonbill  data/394282-adquisicion-cubiertas-maquinarias-vehiculos-utilitarios-annp-1-1621414358.json

file: https://contrataciones.gov.py/datos/api/v3/doc/ocds/record/ocds-03ad3f-394288-1

output: 
1. correct tenders_items table

![image](https://user-images.githubusercontent.com/6834122/119059558-8eff4f00-b99e-11eb-83f8-35506f04e05f.png)

2. incorrect tenders_items columns in tenders table:

![image](https://user-images.githubusercontent.com/6834122/119059626-b3f3c200-b99e-11eb-8559-935a58149c3a.png)

(this may be happening with all the arrays not only items)

**Comment by jpmckinney** [comment]
@yolile What specifically is incorrect? I think we have the logic that if there are <=5 items, then they are by default "rolled up" into the parent table.

**Comment by yolile** [comment]
Yes, but in this case, we have more than 5 items, so the tender_items table is created (correctly) but also all the items related columns are in the parent table (with empty values, incorrectly)

**Comment by yolile** [comment]
@sorenabell is this deployed and fixed in the web interface too?

**Comment by jpmckinney** [comment]
@sorenabell: @yolile tested and this seems to have been fixed in the CLI, but not yet deployed to the web?

**Comment by sorenabell** [comment]
Yes, let us test it a little bit more.

**Comment by sorenabell** [comment]
@yolile @jpmckinney This issue has been fixed in both CLI and web.

**Comment by jpmckinney** [comment]
Great :) So we can close the issue? @yolile can also test again.

**Comment by yolile** [comment]
I tested it again with the attached file and indeed, the problem is fixed for the parent arrays, however, there are some arrays at the item level that appear in both, the parent table (with empty values) and also as a separate table. See, for example "items/attributes": 
- tenders table:
 
![image](https://user-images.githubusercontent.com/6834122/126559424-0edeff10-a768-4598-b715-835eeb6e5208.png)

- tenders_items: (note here that there are also empty columns for the "additionalIdentifiers" array)
 
![image](https://user-images.githubusercontent.com/6834122/126559518-36281346-82fc-4001-8107-ba1eb447ec25.png)

- tenders_items_class and tenders_items_attributes are correct:
![image](https://user-images.githubusercontent.com/6834122/126559689-cd978248-963c-4992-abc1-f6695bb97138.png)
![image](https://user-images.githubusercontent.com/6834122/126559705-199b1354-ad1d-4710-8a5e-6c6e7fffbdad.png)

files:
[result.xlsx](https://github.com/open-contracting/spoonbill/files/6858817/result.xlsx)
[394282-adquisicion-cubiertas-maquinarias-vehiculos-utilitarios-annp-1-1621414358.txt](https://github.com/open-contracting/spoonbill/files/6858821/394282-adquisicion-cubiertas-maquinarias-vehiculos-utilitarios-annp-1-1621414358.txt)

**Comment by sorenabell** [comment]
@yolile a new version of the library has been released that fixes this issue.
Please test this fix on https://ocp12.open-contracting.org/

[results 2021-09-22.xlsx](https://github.com/open-contracting/spoonbill/files/7210996/results.2021-09-22.xlsx)

**Comment by yolile** [comment]
Hmm not sure if I'm doing something wrong, but I tried with this file again https://github.com/open-contracting/spoonbill/files/6858821/394282-adquisicion-cubiertas-maquinarias-vehiculos-utilitarios-annp-1-1621414358.txt and everything looks fine but the tender items. The resulted excel file has only 1 tender item, but the original JSON has 16, and the extra columns are still there in the tender sheet.
[result (7).xlsx](https://github.com/open-contracting/spoonbill/files/7213169/result.7.xlsx)

**Comment by jpmckinney** [comment]
> The resulted excel file has only 1 tender item, but the original JSON has 16

@yolile I'm not sure I follow. `tender_items` has 16 data rows. Spoonbill can't currently merge intermediate tables (i.e. `tenders_items_class` into `tender_items`) so there are also tables for `tenders_items_class` and `tenders_items_attributes`.

> the extra columns are still there in the tender sheet

Yes, the only thing I notice is that `tenders` has many empty columns.

**Comment by yolile** [comment]
> @yolile I'm not sure I follow. tender_items has 16 data rows.

Oh you are right, sorry, I had too many open Excel files and I was probably looking at the incorrect one when I wrote that. You are right, there are 16 items in the excel file, correctly. However, the `tender_items` sheet also has empty columns: `/tender/items/additionalClassifications/0/* `and `/tender/items/attributes/0/*`

**Comment by jpmckinney** [comment]
Hmm, are you sure you are testing your file on ocp12.open-contracting.org? I don't have any empty columns anywhere when I tested.

**Comment by yolile** [comment]
Yes:

https://user-images.githubusercontent.com/6834122/134520020-f1404ae8-bd95-45c5-9e70-025ed79db2bd.mp4

(And I reproduced the issue about having only 1 item now) But if everything is working well for you, then maybe for some reason the problem is just for me and we can close this issue)

**Comment by jpmckinney** [comment]
Aha, at 0:50, I checked "Split arrays into separate tables", in which case everything is fine.

@sabahfromlondon What should be the default state of that toggle?

@yolile I'm able to reproduce empty columns on both the `tenders` and `tenders_items` sheets if I don't check that toggle
[result-2.xlsx](https://github.com/open-contracting/spoonbill/files/7219250/result-2.xlsx)
.
